DBZ-2163 Topic name mapper demo (#115)
debezium/debezium-examples: debezium-server-name-mapper/README.md

# Custom Topic Name Mapper

> This example demonstrates how to develop and deploy a bean that would allow the developer to implement custom topic naming policy.

Reviewer: ```suggestion
This example demonstrates how to develop and deploy a bean that would allow the developer to implement a custom topic naming policy.
```

Reviewer: Should also mention that this is specific to Debezium Server, including a link to its docs.

The implementation of the custom naming policy is in the class `PrefixingNameMapper`.

> The class is marked as `@Dependent` is it is injected into the using sink class.

Reviewer: ```suggestion
The class is marked as `@Dependent` and it is injected into the configured sink adaptor class.
```

> It exposes `mapper.prefix` configuration option that defines the sting that is inserted before the original topic name.

Reviewer: ```suggestion
It exposes a configuration option named `mapper.prefix` which defines a string that is inserted before the original topic name.
```

> The configuration option is configured either via `application.properties` configuration file or via other ways defined in Microprofile Config specification.

Reviewer: ```suggestion
The configuration option is configured either via `application.properties` configuration file or via other ways defined in the Microprofile Config specification, e.g. an environment variable.
```

> As an example - a message intended to be delivered to topic `server.table` would be routed to topic `superprefix.server.table`.

Reviewer: ```suggestion
As an example, when setting `mapper.prefix` to the value `superprefix`, then a message intended to be delivered to the topic `server.schema.table` would be routed to topic `superprefix.server.schema.table`.
```
